baseapp/abci.go

```diff
@@ -266,7 +278,19 @@ func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) abci.Respon
 // Ref: https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-060-abci-1.0.md
 // Ref: https://github.com/tendermint/tendermint/blob/main/spec/abci/abci%2B%2B_basic_concepts.md
 func (app *BaseApp) ProcessProposal(req abci.RequestProcessProposal) abci.ResponseProcessProposal {
-	// TODO: Implement.
+	ctx := app.checkState.ctx
+
+	for _, txBytes := range req.Txs {
+		anteCtx, _ := app.cacheTxContext(ctx, txBytes)
+		tx, err := app.txDecoder(txBytes)
+		if err != nil {
+			return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
```

Comment:
This _reject_ looks right to me, as a TX that fails to decode is a sign that the encoder was buggy, independently of the actual App state

Reply:
Yeah, I agree. If we cannot decode a tx, that indicates some malicious behavior by the proposer in `PrepareProposal`. One thing we have not discussed is how to handle such byzantine behavior. @sergio-mena what will Tendermint do in such a case? Should we instead response with `ACCEPT` and punish the proposer to avoid liveness halts?

Reply:
Failure to decode a tx, or any other error encountered when parsing/validating incoming data, doesn't automatically indicate Byzantine/malicious behavior. It's far more likely to be the result of a bug or fault, either in the code of the participating actors, or at the network layer. You shouldn't punish a proposer if their network provider, or any other intermediary, screws something up! 😅

Reply:
@alexanderbez For the moment, Tendermint will prevent that (malformed) proposal from being decided by prevoting `nil`. Moving forward, we'll see if there is something else to be done. In any case, ProcessProposal "rejects" are a serious condition in a production chain (it indicates a Byzantine behavior, either via a bug, or via a malicious node).

To your second question, you _can_ and _should_ return reject in this case